mobile_verifier/src/heartbeats/mod.rs

```diff
@@ -516,8 +518,43 @@ impl ValidatedHeartbeat {
             }
             GatewayResolution::AssertedLocation(location) if heartbeat.hb_type == HbType::Wifi => {
                 let asserted_latlng: LatLng = CellIndex::try_from(location)?.into();
+                let is_valid = match heartbeat.location_validation_timestamp {
+                    None => {
+                        if let Some(last_location) = last_location_cache
+                            .fetch_last_location(&heartbeat.hotspot_key)
+                            .await?
+                        {
+                            if last_location.lat != 0.0 || last_location.lon != 0.0 {
+                                heartbeat.lat = last_location.lat;
```

Comment:
I get the intent with this but the fact that we are rewriting an ingested heartbeat is not ideal.  Typically we should be outputting ingested reports without modification.  Additional data is usually included as part of a wrapper/envelope.    I get that that using an envelope adds additional complexity and maybe not the best way here, so as an alternative should we consider adding a new field/flag to the heartbeat itself to indicate it has been modified ?

Reply:
thinking further on this, i do think a new proto msg might after all be a good option...could be straight forward to use something like 

modified_heartbeat {
  heartbeat original_report = 1,
  heartbeat modified_report = 2,
}